src/engine/services/time-service.ts

```diff
@@ -0,0 +1,14 @@
+export default class TimeService {
+  static currentFrame: number | undefined;
+
+  private static _delta: number = 0;
+
+  static get delta(): number {
+    return this._delta;
+  }
+
+  static set delta(newDelta: number) {
+    // don't accept negative numbers
+    this._delta = Math.max(newDelta, this._delta);
```

Comment:
This is why you saw the bug you described. The frame rate look liked it dropped when it really didn't. This is a wrong calculation of delta. Once delta hits a slow number, Math.max will lock it at that high value, because it can be set to nothing lower.

TimeServer doesn't really serve a good purpose, and is an unnecessary abstraction. Let's remove it. It is cleaner and makes more sense if GameObject's get the latest delta as a parameter in update. It doesn't really make sense to be accessing it from a static class.